shared/routing.go

```diff
@@ -42,8 +42,14 @@ func WrapHSTS(h http.HandlerFunc) http.HandlerFunc {
 // WrapPermissiveCORS wraps the given handler func in one that sets an
 // all-permissive CORS header on the response.
 func WrapPermissiveCORS(h http.HandlerFunc, methods ...string) http.HandlerFunc {
+	return WrapRestrictiveCORS(h, []string{"*"}, methods)
+}
+
+// WrapRestrictiveCORS wraps the given handler func in one that sets
+// a CORS header with specified origins and methods on the response.
+func WrapRestrictiveCORS(h http.HandlerFunc, origins []string, methods []string) http.HandlerFunc {
 	opts := []handlers.CORSOption{
-		handlers.AllowedOrigins([]string{"*"}),
+		handlers.AllowedOrigins(origins),
```

Comment:
For the use case of triaging metadata, we also need `AllowCredentials` to allow cookies, but we probably don't want that for other requests, so I suggest renaming this method to `WrapTrustedCORS` (because it is not the opposite of `WrapPermissiveCORS` -- it's both more restrictive in terms of origins and more relaxed in terms of credentials), and making `WrapPermissiveCORS` **not** call this method (but duplicating the code instead).